python/cudf/cudf/core/column/column.py

```diff
@@ -682,7 +682,7 @@ def __setitem__(self, key: Any, value: Any):
                     raise ValueError(
                         "Boolean mask must be of same length as column"
                     )
-                key = arange(len(self))[key]
+                key = arange(len(self))[key.fillna(False)]
```

Comment:
Do we need this fillna? What's libcudf behavior when the mask has nulls?

Reply:
We'll error in python before libcudf ever sees it [here in Frame](https://github.com/rapidsai/cudf/blob/branch-0.19/python/cudf/cudf/core/frame.py#L1569-L1571)

```ValueError: cannot mask with boolean_mask containing null values```

Reply:
Should we fix that? What's Pandas behavior when passed a mask with null values?

Reply:
Seems it's treated as `False`:
```
>>> x = pd.Series([1,2,3], dtype='Int64')
>>> x
0    1
1    2
2    3
dtype: Int64
>>> mask = pd.Series([True, False, pd.NA], dtype='boolean')
>>> mask
0     True
1    False
2     <NA>
dtype: boolean
>>> x[mask]
0    1
dtype: Int64
```

Reply:
What about libcudf behavior?

Reply:
So we'd update `__getitem__` and `__setitem__` to `fillna(False)` when presented with a boolmask?

Reply:
Not sure about libcudf behavior - will follow up shortly

Reply:
have not explicitly tested yet but the libcudf header says this about `apply_boolean_mask`:

```
 * Given an input `table_view` and a mask `column_view`, an element `i` from
 * each column_view of the `input` is copied to the corresponding output column
 * if the corresponding element `i` in the mask is non-null and `true`.
 ```
The last line seems to imply that it treats `NULL` as `False`.

Reply:
So is this OK or should we change this somewhere else? I am not sure why our code `__getitem__` our `_apply_boolean_mask` code has this guard in light of the `libcudf` behavior, so I'm a little hesitant to remove it.

Reply:
Lets give it a shot in removing it?

Reply:
Tried this and the only test that popped was [this one](https://github.com/rapidsai/cudf/blob/branch-0.19/python/cudf/cudf/tests/test_indexing.py#L760-L768) that seems to be testing that specific line of code. It seems like the last time this test was touched it was being moved out of `test_dataframe.py` and into `test_indexing.py`, and before that I can't tell where it came from. 

Guessing it's legacy code, but also weary of just removing it...

Reply:
Yup that test looks invalid. We need to make changes to align behaviors, lets do it.